actions/version_number/action.yaml

```diff
@@ -75,11 +69,18 @@ runs:
         if [[ "${{ inputs.version_number }}" != "" ]]; then
           echo "new_version=${{ inputs.version_number }}" >> $GITHUB_OUTPUT
         elif [[ "${{ inputs.type }}" == "Snapshot" ]]; then
-          echo "new_version=$NBGV_SimpleVersion-SNAPSHOT" >> $GITHUB_OUTPUT
+          if [[ -f ${{ inputs.working_dir }}/pom.xml ]]; then
+            MVN_VERSION=$(grep -o '<version>.*</version>' ${{ inputs.working_dir }}/pom.xml | head -1 | sed 's/<version>\(.*\)<\/version>/\1/')
+            IFS='.' read -r -a parts <<< "$MVN_VERSION"
+            PATCH="${parts[2]}"
+            PATCH=$(echo $PATCH | cut -d'-' -f 1)
+            NEW_PATCH=$(($PATCH+1))
+            echo "new_version=$NBGV_MajorMinorVersion.$NEW_PATCH-alpha-$GITHUB_RUN_NUMBER-SNAPSHOT" >> $GITHUB_OUTPUT
+          else
+            echo "new_version=$NBGV_SimpleVersion-SNAPSHOT" >> $GITHUB_OUTPUT
+          fi
         elif [[ "$BRANCH_NAME_TRUNC" != "" && "$BRANCH_NAME_TRUNC" != "master" && "$BRANCH_NAME_TRUNC" != "main" ]]; then
-          echo "new_version=$NBGV_SimpleVersion-$NBGV_GitCommitIdShort-$BRANCH_NAME_TRUNC" >> $GITHUB_OUTPUT
-        elif [[ "${{ inputs.short_name }}" == "Yes" ]]; then
-          echo "new_version=$NBGV_SimpleVersion" >> $GITHUB_OUTPUT
+          echo "new_version=$NBGV_SimpleVersion-$BRANCH_NAME_TRUNC" >> $GITHUB_OUTPUT
```

Comment:
What cases do we have with no version supplied and no `pom.xml` in the working dir? It looks like there's a risk for branch builds in that case - we'll go from having uniquely-identified branch releases to having conflicting ones for multiple releases on the same branch.

Reply:
Actually - I wasn't accounting for NBGV incrementing the version. I think we're fine.